core/src/indexer/fetch_logs.rs

```diff
@@ -36,99 +39,271 @@ pub fn fetch_logs_stream(
     let contract_address = initial_filter.contract_address();
 
     tokio::spawn(async move {
-        let snapshot_to_block = initial_filter.get_to_block();
-        let from_block = initial_filter.get_from_block();
-        let mut current_filter = initial_filter;
-
-        // add any max block range limitation before we start processing
-        let mut max_block_range_limitation =
-            config.network_contract.cached_provider.max_block_range;
-        if max_block_range_limitation.is_some() {
-            current_filter = current_filter.set_to_block(calculate_process_historic_log_to_block(
-                &from_block,
-                &snapshot_to_block,
-                &max_block_range_limitation,
-            ));
-            warn!(
+        // if we have a, exex backfill tx, we need to process the exex stream instead of using the
+        // rpc provider
+        if let Some(backfill_tx) = backfill_tx {
```

Comment:
maybe there's a better way to check if exex mode is enabled. checking if backfill_tx exists is quite implicit